src/toil/bus.py

```diff
@@ -599,5 +623,84 @@ def __init__(self) -> None:
         """
         super().__init__()
 
+    def _set_bus_and_message_types(self, bus: MessageBus, wanted_types: List[type]) -> None:
+        """
+        Connect to the given bus and collect the given message types.
+
+        We must not have connected to anything yet.
+        """
+
+        # We need to call the two inherited connection methods
+        super()._set_bus_and_message_types(bus, wanted_types)
+        self._set_bus(bus)
 
 
+def replay_message_bus(path: str) -> Dict[str, Any]:
```

Comment:
This should have a better type signature, or else a *very* clear description of what exactly we return and what the caller is going to be allowed to do with it.

Maybe to do that we want to promote `JobStatus` out of the function?

Reply:
Maybe later but I think job status overlaps with with how toil thinks of job status internally already and I don't think there's a need to promote it outside.